src/hooks/useSelectChain.ts

```diff
@@ -21,13 +24,14 @@ export default function useSelectChain() {
 
       try {
         await switchChain(connector, targetChain)
+        if (isSupportedChain(targetChain)) setSearchParams({ chain: CHAIN_IDS_TO_NAMES[targetChain] })
```

Comment:
Should this retain other search params?

```suggestion
        if (isSupportedChain(targetChain)) {
          setSearchParams((searchParams) => ({
            ...searchParams,
            chain: CHAIN_IDS_TO_NAMES[targetChain]
          }))
        }
```